### PR TITLE
[CC-15] Add django-distill to project

### PR DESCRIPTION
**Pull request by Audiosutras**
## Description

This PR adds the `django-distill` package to the project so that we are able to output static files based upon our `licenses.urls` url endpoints.

## Technical details

```
python manage.py distill-local --collectstatic 
```
This command will `build` static files locally. 
```
python manage.py distill-local --force
```
When changes to the code occur and you want to run a new build, you can use the `--force` command to tell django that it is ok to overwrite the files that currently in the build directory.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: i18n/utils.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
-from babel import Locale, UnknownLocaleError
+from babel import Locale
+from babel import UnknownLocaleError
```
Comment (dpoirier) [inline i18n/utils.py:2]: We usually group imports from the same package into a single import statement. I'd like to continue with that style if we can.

File: licenses/tests/test_utils.py [hunk 2]
```diff
@@ -152,3 +157,99 @@ def test_get_license_url_from_legalcode_url(self):
             get_license_url_from_legalcode_url(
                 "http://opensource.org/licences/bsd-license.php"
             )
+
+
+class GetLicenseUtilityTest(TestCase):
+    """Test django-distill utility functions for
+    generating an iterable of license dictionaries
+    """
+
```
Comment (dpoirier) [inline licenses/tests/test_utils.py:166]: With CC-8, we'll no longer have licenses in the database when tests start. Which is probably just as well. We'll need to create the licenses we want for each test using e.g. LicenseFactory.

(I don't think these tests will fail, but they won't really be testing anything if there are no licenses.)

File: i18n/utils.py (hunk 1 above)
Comment (Audiosutras) [reply i18n/utils.py:2]: We currently have both `pre-commit` and `make` file linting and testing on the repo. This is causing formatting conflicts. @dpoirier

Comment (Audiosutras) [reply i18n/utils.py:2]: https://github.com/asottile/reorder_python_imports

Comment (Audiosutras) [reply i18n/utils.py:2]: Removed `pre-commit` from the repository. `Makefile` is now the source of truth.

Comment (Audiosutras) [reply i18n/utils.py]: done. re-ordered imports

File: licenses/tests/test_utils.py (hunk 2 above)
Comment (Audiosutras) [reply licenses/tests/test_utils.py:170]: Imported `LicenseFactory` and added licenses to `GetLicenseUtilityTest`.

File: licenses/tests/factories.py [hunk 3]
```diff
@@ -2,12 +2,11 @@
...
-    License,
-    LegalCode,
-    TranslatedLicenseName,
-    LicenseLogo,
-)
+from licenses.constants import LICENSE_VERSIONS
+from licenses.models import LegalCode
+from licenses.models import License
+from licenses.models import LicenseLogo
+from licenses.models import TranslatedLicenseName
```
Comment (dpoirier) [inline licenses/tests/factories.py:9]: Can we combine these import statements?

File: licenses/tests/test_models.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 
 from i18n import DEFAULT_LANGUAGE_CODE
-from licenses import FREEDOM_LEVEL_MIN, FREEDOM_LEVEL_MID, FREEDOM_LEVEL_MAX
-from licenses.models import (
-    LegalCode,
-    License,
-    LicenseLogo,
-    TranslatedLicenseName,
-)
+from licenses import FREEDOM_LEVEL_MAX
```
Comment (dpoirier) [inline licenses/tests/test_models.py:5]: Can we combine these into two import statements?

File: licenses/urls.py [hunk 5]
```diff
@@ -136,33 +143,32 @@ def to_url(self, value):
...
-    ),
-    path(
-        "deed/", deed_detail, name="deed_detail"
-    ),
-    path(
-        "<code:license_code>/<version:version>",
+    # New templates to be used
+    path("license/", license_detail, name="license_detail"),
+    path("sampling/", sampling_detail, name="sampling_detail"),
+    path("deed/", deed_detail, name="deed_detail"),
```
Comment (dpoirier) [inline licenses/urls.py:149]: It looks like we could change `path` to `distill_path` here and add a no-op distill_func and we'd get our test pages included in the static site.

File: licenses/tests/factories.py (hunk 3 above)
Comment (Audiosutras) [reply licenses/tests/factories.py:9]: done

File: licenses/tests/test_models.py (hunk 4 above)
Comment (Audiosutras) [reply licenses/tests/test_models.py:5]: done

File: licenses/tests/factories.py (hunk 3 above)
Comment (Audiosutras) [reply licenses/tests/factories.py]: done

File: licenses/urls.py (hunk 5 above)
Comment (Audiosutras) [reply licenses/urls.py]: done

**Review by dpoirier — APPROVED** [review]

**Pull request merged**